tools/global_setup.sh

```diff
@@ -6,7 +6,7 @@ echo "Starting setup..."
 # handle different python alias
 echo "Looking through possible python aliases"
 
-REQUIRED_VERSION=3.9
+REQUIRED_VERSION=3.11
```

Comment:
those are ok to keep at >3.9
since the 3.9 is supported until 2025
Please update only the github actions to make the tests go faster, since 3.11 introduces performance improvements 
https://www.python.org/downloads/